.github/workflows/deploy-to-pantheon.yml

```diff
@@ -17,114 +17,24 @@ on:
-      - name: install gh cli
-        run: |
-          type -p curl >/dev/null || (sudo apt update && sudo apt install curl -y)
-          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg \
-          && sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg \
-          && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
-          && sudo apt update \
-          && sudo apt install gh -y
-      - name: Get default branch
-        id: get-origin-default-branch
-        run: |
-          export BRANCH=`gh api /repos/${{ github.repository }} --jq ".default_branch"`
-          echo "default_branch=${BRANCH}" >> "$GITHUB_OUTPUT"
-
-      - name: Cache bash_env.txt
-        uses: actions/cache@v4
-        env:
-          cache-name: cache-bash-env
-        with:
-          path: bash_env.txt
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ github.run_number }}-${{ github.run_attempt }}
-
-      # Set TERMINUS_ENV and related environment variables.
-      # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
-      - name: setup-environment-vars
-        run: |
-          export CI_PROJECT_REPONAME=`curl -u "${GITHUB_OWNER}:${GITHUB_TOKEN}" \
-            -H "Accept: application/vnd.github.groot-preview+json" \
-            "https://api.github.com/repos/${CI_PROJECT_NAME}/commits/${COMMIT_SHA}/pulls" | \
-            python3 -c "import sys, json; print(json.load(sys.stdin)[0]['base']['repo']['name'])"`
-          export CI_PROJECT_USERNAME=$GITHUB_REPOSITORY_OWNER
-          export DEFAULT_BRANCH='${{ steps.get-origin-default-branch.outputs.default_branch}}'
-          /build-tools-ci/scripts/set-environment
-          GITHUB_WORKFLOW_URL=https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID
-          echo "export CI_BUILD_URL='${GITHUB_WORKFLOW_URL}'" >> $BASH_ENV
-          echo "export CI_NODE_INDEX=0" >> $BASH_ENV
-          echo "export CI_REPOSITORY_URL='https://github.com/${GITHUB_REPOSITORY}'" >> $BASH_ENV
-          echo "export ARTIFACTS_DIR_URL='${GITHUB_WORKFLOW_URL}/#artifacts'" >> $BASH_ENV
 # important note: deploy_to_pantheon will push EVERYTHING in the repo directory other than the .git directory, i.e.
 # it will not respect .gitignore
   deploy_to_pantheon:
       name: "Deploy to Pantheon"
-      container:
-        image: quay.io/pantheon-public/build-tools-ci:6.x
-        options: --user root
       runs-on: ubuntu-latest
-      needs: [configure_env_vars]
+      permissions:
+        deployments: write
+        contents: read
+        pull-requests: read
+
       steps:
         - name: Checkout
           uses: actions/checkout@v4
-          with:
-            fetch-depth: 0
```

Comment:
The new github action doesn't like `fetch-depth: 0` and will blow up. We don't normally use that in our checkouts so we should be fine removing it.